starknetdata/starknetdata.go

```diff
@@ -11,7 +11,7 @@ import (
 //
 //go:generate mockgen -destination=../mocks/mock_starknetdata.go -package=mocks github.com/NethermindEth/juno/starknetdata StarknetData
 type StarknetData interface {
-	BlockByNumber(ctx context.Context, blockNumber uint64) (*core.Block, error)
+	BlockByID(ctx context.Context, blockID string) (*core.Block, error)
```

Comment:
We lose type safety here for no real reason. Let's keep it the old way and introduce a `Head` or a similar endpoint